django/librarian/utils/markdown_splitter.py

```diff
@@ -7,12 +7,15 @@
 
 
 class MarkdownSplitter:
-    def __init__(self, chunk_size=768, chunk_overlap=100, debug=False):
+    def __init__(
+        self, chunk_size=768, chunk_overlap=100, enable_markdown=True, debug=False
+    ):
         self.chunk_size = chunk_size
         self.chunk_overlap = chunk_overlap
         self.debug = debug
         self.current_headings = {i: None for i in range(1, 7)}
         self.last_table_header = None
+        self.enable_markdown = enable_markdown
```

Comment:
Users may upload plain text files or PDFs that have markdown-looking content (lines starting with #, etc) but we know that they aren't actual markdown. In those cases we don't want to do any special processing based on markdown syntax.

Reply:
very good point